### PR TITLE
MINOR: System test ZooKeeper upgrades

### PR DESCRIPTION
**Pull request by rondagostino**

### CODE REVIEW

**Comment by rondagostino** [comment]
@omkreddy Could you please review?  The zstd upgrade system tests failed locally but also failed on trunk, so I assume that's a config issue on my end.  I also ran the TLS system tests locally, and they passed.

File: tests/kafkatest/tests/core/upgrade_test.py [hunk 1]
```diff
@@ -49,6 +47,9 @@ def wait_until_rejoin(self):
                     backoff_sec=1, err_msg="Replicas did not rejoin the ISR in a reasonable amount of time")
 
     def perform_upgrade(self, from_kafka_version, to_message_format_version=None):
+        self.logger.info("Upgrade ZooKeeper from %s to %s" % (str(self.zk.nodes[0].version), str(DEV_BRANCH)))
+        self.zk.set_version(DEV_BRANCH)
+        self.zk.restart_cluster()
```
Comment (omkreddy) [inline tests/kafkatest/tests/core/upgrade_test.py:52]: looks like the test is failing at below `self.kafka.stop_node(node)` call, without `snapshot.trust.empty` config.  Can we check for topic/znode existence in ZK after upgrade. We can use ZK cli/kafka-topic describe? This way we can test ZK server availability and znode  data existence after upgrade.

**Review by omkreddy — COMMENTED** [review]
@rondagostino Thanks for PR. LGTM. left one comment.

File: tests/kafkatest/tests/core/upgrade_test.py (hunk 1 above)
Comment (rondagostino) [reply tests/kafkatest/tests/core/upgrade_test.py:52]: All set -- good point about making the failure look more intuitive by trying to describe the topic before doing the Kafka roll.  I confirmed the failure now happens on the describe topic command when leaving off the `snapshot.trust.empty=true` config, and I confirmed the tests still succeed when that config is left in.

**Review by omkreddy — APPROVED** [review]
@rondagostino Thanks for the update. LGTM.

**Pull request merged**